mimic/rest/nova_api.py

```diff
@@ -399,6 +413,44 @@ def perform_action(self, request, tenant_id, server_id):
         """
         return self._region_collection_for_tenant(tenant_id).request_action(request, server_id, self.url)
 
+    @app.route("/v2/<string:tenant_id>/os-keypairs", methods=['GET'])
+    def get_key_pairs(self, request, tenant_id):
+        """
+        Returns current key pairs
+        """
+        return self._keypair_collection_for_tenant(tenant_id).json_list()
+
+    @app.route("/v2/<string:tenant_id>/os-keypairs", methods=['POST'])
+    def create_key_pair(self, request, tenant_id):
+        """
+        Returns a newly created key pair with the specified name.
+        """
+        try:
+            content = json.loads(request.content.read())
+            keypair = content["keypair"]
+            keypair_from_request = KeyPair(name=keypair["name"], public_key=keypair["public_key"])
```

Comment:
That route isn't being covered in this PR since its a rare use case(asking for the API to generate a pub/private key pair for you) since most of the time end users are uploading there own public key.

The route I'm mocking is here:
http://docs.rackspace.com/servers/api/v2/cs-devguide/content/UploadKeyPair.html

The api generating  a keypair for you route was out of the scope of this pull-request and doesn't have a use case for mimic yet so it hasn't been included yet.

I will update this pull-request to include a fake `user_id` though.